target/linux/ramips/image/Makefile

```diff
@@ -28,6 +28,12 @@ LOADER_FLASH_START := $(ldrflashstart-y)
 
 KERNEL_DTB = kernel-bin | append-dtb | lzma
 
+define Build/humax-factory
+	mv $@ upg
+	zip -P $(1) $@.new upg
```

Comment:
I guess `zip -P $(1) $@.new $@` would suffice here?

Reply:
No. stock 'WebUpgrade' function are checks filename inside archive, and It's must be 'upg'.